client/src/components/aggregations/utils.js

```diff
@@ -87,15 +88,15 @@ export const countPerValueAggregation = (fieldName, fieldConfig, data) => {
     return counter
   }, {})
   const legendColors = _clone(CHART_COLORS)
-  const legend = fieldConfig?.choices || {}
+  const legend = _cloneDeep(fieldConfig?.choices || {})
```

Comment:
wouldn't this line change have been sufficient? Is there a reason to change the rest of the code?

Reply:
There is a nuance for which if `counters` have `label` and color `fields`, previous version ( Object.entries version) would have different behavior than Object.keys version ( i missed this when i refactored to Object.entries). I just wanted to preserve the initial logic and only add cloning.